static/app/components/avatar/avatarList.tsx

```diff
@@ -43,16 +45,31 @@ function AvatarList({
           </CollapsedUsers>
         </Tooltip>
       )}
-      {visibleUsers.map(user => (
-        <StyledAvatar
-          key={`${user.id}-${user.email}`}
-          user={user}
-          size={avatarSize}
-          renderTooltip={renderTooltip}
-          tooltipOptions={tooltipOptions}
-          hasTooltip
-        />
-      ))}
+      {visibleMembers.map(member => {
+        if (member.type && member.type === 'team') {
+          const team = member as TeamParticipant;
```

Comment:
You shouldn't need to cast this here, `member.type === 'team'` should be enough for typescript to discriminate correctly. Maybe it's because you have `type` being optional?

https://www.typescriptlang.org/play?#code/C4TwDgpgBAqgzhATlAvFA3gKCjqBLAEwC4o5hE8A7AcwG5tdRISByAVwURcwF9NMm0ACoQAhgFtUGBjkoSIJMhRoyog1sDHjufAeGgAFUYmB4AxnjCjKwOFICCiRKJAAeERKgAfWJwB8-GYA9pRkUFYm5pbWtiRGkRZWNnZoANoAuvwRpokxcAB0AGZBiACiomYAFgAUYKh+0rj4hVC1+YKoKGjsnCwAlI1NuGD5hKo8UBAANgiDQ+H5cuIQ47x9QA

Reply:
Oh I see, it's because AvatarTeam doesn't have a `type`. A bit confusing that UserParticipant isn't used here, but I can see you probably did it because there are many uses right now that would break?

Reply:
I think my recommended here would be to either:

1. Add `type: 'user'` to the user type here to make it consistent, or
2. Separate `users` and `teams` instead of the combined `members`. But if order matters and you need teams and users to be interwoven this wouldn't be a good solution.